crates/blockifier/src/execution/stack_trace_test.rs

```diff
@@ -84,66 +86,85 @@ fn test_stack_trace(
     )
     .unwrap_err();
 
-    // Fetch PC locations from the compiled contract to compute the expected PC locations in the
-    // traceback. Computation is not robust, but as long as the cairo function itself is not edited,
-    // this computation should be stable.
-    let account_entry_point_offset =
-        account.get_entry_point_offset(selector_from_name(EXECUTE_ENTRY_POINT_NAME));
     let execute_selector_felt = selector_from_name(EXECUTE_ENTRY_POINT_NAME).0;
-    let external_entry_point_selector_felt = selector_from_name(call_contract_function_name).0;
-    let entry_point_offset =
-        test_contract.get_entry_point_offset(selector_from_name(call_contract_function_name));
-    // Relative offsets of the test_call_contract entry point and the inner call.
-    let call_location = entry_point_offset.0 + 14;
-    let entry_point_location = entry_point_offset.0 - 3;
-    // Relative offsets of the account contract.
-    let account_call_location = account_entry_point_offset.0 + 18;
-    let account_entry_point_location = account_entry_point_offset.0 - 8;
-
-    let expected_trace_cairo0 = format!(
-        "Transaction execution has failed:
+
+    let expected_trace = match cairo_version {
+        CairoVersion::Cairo0 | CairoVersion::Cairo1 => {
+            // Fetch PC locations from the compiled contract to compute the expected PC locations in
+            // the traceback. Computation is not robust, but as long as the cairo
+            // function itself is not edited, this computation should be stable.
+            let account_entry_point_offset =
+                account.get_entry_point_offset(selector_from_name(EXECUTE_ENTRY_POINT_NAME));
+            let external_entry_point_selector_felt =
+                selector_from_name(call_contract_function_name).0;
+            let entry_point_offset = test_contract
+                .get_entry_point_offset(selector_from_name(call_contract_function_name));
+            // Relative offsets of the test_call_contract entry point and the inner call.
+            let call_location = entry_point_offset.0 + 14;
+            let entry_point_location = entry_point_offset.0 - 3;
+            // Relative offsets of the account contract.
+            let account_call_location = account_entry_point_offset.0 + 18;
+            let account_entry_point_location = account_entry_point_offset.0 - 8;
+
+            let expected_trace_cairo0 = format!(
```

Comment:
Why do you create both trace expectations. What do you think of just doing

```
if Cairo0 {
  return Cairo0 expected trace
} 

return Cairo 1 expected trace
```
The code seems more readable to me that way

Reply:
We have match expr with Native here as well, so I wanted to make it consistent, and also it will help in the future if the new versions of enum will be added, so we don't forget about it